lighthouse-core/lib/network-recorder.js

```diff
@@ -270,18 +294,66 @@ class NetworkRecorder extends EventEmitter {
     }
   }
 
+  /**
+   * @param {string} requestId
```

Comment:
comment on what a "real" request is?

For my own confusion, the IDs of a chain of redirected requests are all the same? Hence the `redirectDestination` chain traversal? And the assumption is that if there's more data for a request it can only be the last in the redirect chain because the rest are finished?

Reply:
> For my own confusion, the IDs of a chain of redirected requests are all the same?

Yup

> And the assumption is that if there's more data for a request it can only be the last in the redirect chain because the rest are finished?

Yup, after another `requestWillBeSent` is issued signaling a redirect, that is the final message of the original request and the rest are about the redirected request.

Reply:
added a comment basically saying this 👍